### ISSUE TITLE
cern-it-sdc-id/dmc/davix fails to build

### ISSUE THREAD

**Issue by maarten-k** [cern-fts/homebrew-dmc#2]
davix fails to build under Mac OS el capitan. Build hangs after:

> -- Checking for module 'gsoapssl++'
> --   Found gsoapssl++, version 2.8
> --  - wsdlh : /usr/local/bin/wsdl2h
> --  - SOAPCPP2 : /usr/local/bin/soapcpp2 

log output can be find in  [log.txt](https://github.com/cern-it-sdc-id/homebrew-dmc/files/109393/log.txt)

**Comment by gbitzes** [comment]
I forgot about this one, my bad.. I'll push a fix on Monday, cheers

**Comment by gbitzes** [comment]
Hi, the dependency on OpenMP has been removed. Could you confirm that it builds now?

**Comment by maarten-k** [comment]
@gbitzes   Where can I find the fix/patch?

**Comment by gbitzes** [comment]
I committed the fix to both master and devel branches of the davix repo. I'm not familiar with Mac OS, but I think that running the homebrew recipe should work now, since it appears it pulls the code directly from git.

**Comment by maarten-k** [comment]
@gbitzes  I thought there was a change in the Formula, but i did  not check the source of the software itself:

However, It still freezes with the same line(see 
[log2.txt](https://github.com/cern-it-sdc-id/homebrew-dmc/files/111507/log2.txt)
)

**Comment by gbitzes** [comment]
Hi, which command are you running? I do "brew install --build-from-source cern-it-sdc-id/dmc/davix" and it works.

**Comment by maarten-k** [comment]
I used 

`$brew install --build-from-source --verbose cern-it-sdc-id/dmc/davix`

which gave me the same results. Since it stoped
Since it stopped at SOAPCPP2 , likely the error is there: I am using version gsoap 2.8.25

**Comment by gbitzes** [comment]
You were right, there was still one more issue to be fixed. It seems that the soapcpp2 tool used when configuring with cmake has an incompatibility between the Linux and MacOS versions.

In Linux, running "soapcpp2 -v" gives you version info, but on MacOS it activates verbose output, and the tool hangs waiting for input.

Try again

**Comment by maarten-k** [comment]
Nice! It did work. However the  cgsi-gsoap package runs into the same problem. Shall I open a separate issue?

**Comment by gbitzes** [comment]
Hi Maarten,

I fixed the issue in cgsi-gsoap, too. It was using the same code to find the version of soapcpp2. However, I'm now getting a different error for this package, "Could not find voms". Are you getting the same?

I'm not sure if this is related to differences in voms2 / voms3, I'll have to ask Alex.

**Comment by maarten-k** [comment]
The voms packages is now the voms-clients 3.0.6 which are based on java and do not have the libraries as before. I think reverting to voms 2.0 without the  voms client tools and a separate voms-clients formula for the current formula would be a solution. (or make a separate voms-libs packages?)

**Comment by gbitzes** [comment]
I see. I made cgsi-gsoap depend on voms2, and for the time being I'm using the old recipe from homebrew to build it.

Now the package should build, I think. You might have to remove the voms package, though, since the two versions conflict, even though they have different names. (voms2 vs voms)

**Comment by maarten-k** [comment]
I am able to install davix , gfal2 gfal2-utilities smoothly.

However, I do have some problems with importing the python module:

> > > /System/Library/Frameworks/Python.framework/Versions/2.7/bin/python
> > > Python 2.7.10 (default, Oct 23 2015, 18:05:06)
> > > [GCC 4.2.1 Compatible Apple LLVM 7.0.0 (clang-700.0.59.5)] on darwin
> > > Type "help", "copyright", "credits" or "license" for more information.
> > > import gfal2
> > > Traceback (most recent call last):
> > >   File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site.py", line 550, in <module>
> > >     main()
> > >   File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site.py", line 532, in main
> > >     known_paths = addusersitepackages(known_paths)
> > >   File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site.py", line 266, in addusersitepackages
> > >     user_site = getusersitepackages()
> > >   File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site.py", line 241, in getusersitepackages
> > >     user_base = getuserbase() # this will also set USER_BASE
> > >   File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site.py", line 231, in getuserbase
> > >     USER_BASE = get_config_var('userbase')
> > >   File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/sysconfig.py", line 520, in get_config_var
> > >     return get_config_vars().get(name)
> > >   File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/sysconfig.py", line 453, in get_config_vars
> > >     import re
> > >   File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/re.py", line 108, in <module>
> > >     import _locale
> > > SystemError: dynamic module not initialized properly``

**Comment by maarten-k** [comment]
My bad!  I used a different version than compiling python which results in this error.

> [15:56:19]maartenk@macbook/usr/local/Cellar/python/2.7.11/bin
> $./python
> Python 2.7.11 (default, Feb 22 2016, 22:46:52)
> [GCC 4.2.1 Compatible Apple LLVM 7.0.2 (clang-700.1.81)] on darwin
> Type "help", "copyright", "credits" or "license" for more information. import gfal2

Thanks for helping me out!
